lib/asyncapiSchemaFormatParser.js

```diff
@@ -11,10 +12,13 @@ module.exports = {
 /**
  * @private
  */
-async function parse({ message, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload }) {
+async function parse({ message, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload, defaultSchemaFormat }) {
   const payload = message.payload;
   if (!payload) return;
 
+  message['x-parser-original-schema-format'] = message.schemaFormat || defaultSchemaFormat;
+  message['x-parser-original-payload'] = cloneDeep(message.payload);
```

Comment:
🤔 I have the feeling that this deep cloning is going to cause performance issues, especially on big payload definitions. It would be great to see how much performance degrades after this PR.

Reply:
Yep! This was raised as part of the issue that got us here 😅  If it does start to cause issues, it would still be nice to provide a way to optionally do a deep clone? That way users can decide if a schema is too big for them to care about keeping the original.

Reply:
Yeah 😅 
We discussed it and in the end, I was like, yeah, we do not do any performance tests and first, we need to do it [here](https://github.com/asyncapi/parser-js/issues/252) I think, and was thinking this should not block this change 🤷🏼‍♂️

Reply:
Not fully convinced but I don't want to be a blocker here. It's true that we don't have performance tests but adding things on purpose knowing it's going to heavily affect performance doesn't make me happy either 😅  Anyway, let's just move forward and we'll work on performance improvements later. I also don't want to unnecessarily add complexity to the API to optionally clone the payload. Thanks for clarifying!

Reply:
@fmvilas 
I'm not 100% sure either, I just tested it manually on local with `console.time` with current tests (yes they are small files, I know) and such deep close was not really adding too much.

```
  console.time('time to deep clone');
  message['x-parser-original-payload'] = cloneDeep(message.payload);
  console.timeEnd('time to deep clone');
```

This is what I was getting:
```
time to deep clone: 0.064ms
    ✓ should parse YAML
time to deep clone: 0.052ms
    ✓ should parse YAML correctly when no components object
time to deep clone: 0.062ms
time to deep clone: 0.043ms
    ✓ should parse 2 AsyncAPI specs in Promise.all() and not fail with resolving references (54ms)
time to deep clone: 0.036ms
    ✓ should apply traits to messages even with empty channels object
time to deep clone: 0.039ms
    ✓ should apply traits to messages used and not used in a channel
time to deep clone: 0.035ms
    ✓ Should include schemas after circular property
time to deep clone: 0.071ms
time to deep clone: 0.024ms
time to deep clone: 0.021ms
time to deep clone: 0.046ms
    ✓ should properly mark circular references
```

The schema that took the longest was https://github.com/asyncapi/parser-js/blob/master/test/good/circular-refs.yaml#L71 which is pretty complex I would say, and, still if we have 1000 schemas like that, it would only add 71ms to parsing.

This is why I was like: "yeah, not much harm will be done I guess, let's try"